spec/pagseguro/extensions/collection_object_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+describe PagSeguro::Extensions::CollectionObject do
+  before(:all) do
+    # Collection
+    class MyItems
+      include PagSeguro::Extensions::CollectionObject
+    end
+
+    # Item
+    class MyItem
+      attr_accessor :name
+
+      def ==(other)
+        name.eql?(other.name)
+      end
+    end
+  end
+
+  let(:collection) do
+    MyItems.new
+  end
+
+  let(:object) do
+    item = MyItem.new
+    item.name = 'Item1'
+    item
+  end
+
+  context "include module" do
+    it "Enumerable" do
+      expect(described_class.included_modules).to include(Enumerable)
+    end
+
+    it "PagSeguro::Extensions::EnsureType" do
+      expect(described_class.included_modules).to include(PagSeguro::Extensions::EnsureType)
+    end
+  end
+
+  context "when was included in a new class" do
+    it "new class must be include PagSeguro::Extensions::CollectionObject" do
+      expect(MyItems.included_modules).to include(PagSeguro::Extensions::CollectionObject)
+    end
+
+    context "and when this klass is instantiated" do
+      it "must be initialized empty" do
+        expect(collection).to be_empty
+      end
+
+      context "must respond to" do
+        [
+          :size,
+          :clear,
+          :empty?,
+          :any?,
+          :each,
+          :include?,
+          :<<
+        ].each do |method|
+          it "#{method}" do
+            expect(collection).to respond_to(method)
+          end
```

Comment:
I thought how is a module, the main function is provide some methods to a class. Perhaps don't make sense verify this in the class that will include this module, because there is very useful I test the behaviour, but I think this is relevant, even because I use *def_delegator* and this is a way to test it.